### ISSUE TITLE
SCSS parent selector (&) causes false positive unused CSS warnings

### ISSUE THREAD

**Issue by mcantonbul** [malinindev/check-unused-css#61]
### Description
First of all, thank you for this amazing tool! It's been really helpful.

I noticed that the tool incorrectly reports nested selectors using SCSS parent reference (`&`) as unused CSS. It fails to recognize that `&Suffix` syntax compiles to `parentSuffix` class name.

### Steps to Reproduce
1. Run `npm run check-unused-css`
2. Check the output for files with nested SCSS selectors

### Expected Behavior
The tool should recognize that nested selectors with `&` compile to combined class names and are being used in components.

### Actual Behavior
The tool reports these compiled class names as unused CSS.

### Code Example

**Component (ProductCard.tsx):**
```tsx
<div className={style.productCard}>
  <h3 className={style.productCardTitle}>Product Name</h3>
  <button className={style.productCardButton}>Add to Cart</button>
</div>
```
**CSS File (ProductCard.module.scss)**
```scss
.productCard {
  padding: 16px;
  
  &Title {  // Compiles to .productCardTitle
    font-size: 18px;
  }
  
  &Button {  // Compiles to .productCardButton
    background: blue;
  }
}
```
Tool Output:

Found 2 classes used in TypeScript but non-existent in CSS:

productCardTitle
productCardButton

**Comment by malinindev** [comment]
Thanks for the report! Glad the tool is useful to you
Fixed in v0.2.7. & concatenation (&Title, &__element, &--modifier) should now resolve correctly.

Let me know if anything's still off
